ROADMAP.md

```diff
@@ -0,0 +1,187 @@
+- Make the tool more intuitive for new users.
+- Make the tool more useful for navigating models with deep hierarchies and large number of 
+  components/variables.
+
+### Potential Challenges:
+- The current user interface is stretched to its limit, and can't integrate any expanded 
+  navigational features. We'll need a new concept for the UI.
+
+### Notes:
+- A POEM detailing proposed new API will be posted by end of January 2020 detailing proposed new API concept
+
+## 3) Integrate N<sup>2</sup> with CaseRecorder data so the state of specific parts of the model can be inspected via case DBs. 
+### Goal: 
+- Leverage the N<sup>2</sup> interface to allow users to understand their models more efficiently.
+- Potentially useful as a debugging tools.
+
+### Potential Challenges: 
+- It is unclear how will this functionality should be implemented. 
+  Current possibilities are Jupyter Lab, or stand alone Electron based javascript app. 
+
+
+----------------------------------------------
+----------------------------------------------
+----------------------------------------------
+# New Differentiation Tools that Reduce User Effort
+
+**Motivation:** While analytic derivatives provide massive performance gains, they also require 
+significant user effort to implement. This creates a high activation energy that prevents many 
+users from taking advantage of the most powerful feature of OpenMDAO. 
+
+**Overall Goal:** Lower the effort required to implement analytic partial derivatives for components, 
+offering a spectrum of options that can potentially trade required user effort with computational efficiency. 
+
+## 1) Coloring applied to approximated partial derivatives 
+### Goal: 
+- Make coloring usable for FD and CS approximations.
+- Offer much higher performance for components with very sparse partial derivative Jacobians 
+  (e.g. vectorized components with no interaction between vector elements).
+- When used with CS, this effectively offers a fast-forward mode AD.
+
+### Potential Challenges: 
+- The computational cost of coloring may be excessive for models with a lot of instances.
+- Coloring algorithms are not perfect and may result in an invalid Jacobian, 
+  so we need an effective way for users to check their colored partials. 
+  The existing `check_partials` functionality can be leveraged, but may need some updates.
+- In some cases, sparsity may not be high enough to offer meaningful performance gains. 
+  We need to provide a means for the user to check this.
+
+### Notes: 
+- A prototype implementation is already available as an experimental feature.
+- The partial coloring algorithm will also be useful for AD partials.
+
+
+## 2) Algorithmic differentiation for component partial derivatives
+    
+### Goal: 
+- Provide forward and reverse mode AD that works for a wide variety of general use cases including
+  with many numpy functions.
+- AD should have relatively good performance compared to hand differentiation.
+
```

Comment:
compared -> comparable  (i assume you are saying that AD will almost be as fast as hand differentiation, not that AD will be faster.)